x-pack/plugins/apm/server/lib/transaction_groups/fetcher.ts

```diff
@@ -36,66 +38,159 @@ interface TopTraceOptions {
 export type Options = TopTransactionOptions | TopTraceOptions;
 
 export type ESResponse = PromiseReturnType<typeof transactionGroupsFetcher>;
+
+export type TransactionGroupRequestBase = ESSearchRequest & {
+  body: {
+    aggs: {
+      transaction_groups: Unionize<
+        Pick<AggregationOptionsByType, 'composite' | 'terms'>
+      >;
+    };
+  };
+};
+
+export type TransactionGroupSetup = Setup & SetupTimeRange & SetupUIFilters;
+
 export async function transactionGroupsFetcher(
   options: Options,
-  setup: Setup & SetupTimeRange & SetupUIFilters,
+  setup: TransactionGroupSetup,
   bucketSize: number
 ) {
-  const { client } = setup;
-
   const projection = getTransactionGroupsProjection({
     setup,
     options,
   });
 
-  const sort: SortOptions = [
-    { _score: 'desc' as const }, // sort by _score to ensure that buckets with sampled:true ends up on top
-    { '@timestamp': { order: 'desc' as const } },
-  ];
-
   const isTopTraces = options.type === 'top_traces';
 
-  if (isTopTraces) {
-    // Delete the projection aggregation when searching for traces, as it should use the combined aggregation instead
-    delete projection.body.aggs;
-  }
+  delete projection.body.aggs;
+
+  const size = bucketSize + 1;
 
-  const params = mergeProjection(projection, {
+  const request = mergeProjection(projection, {
+    size: 0,
     body: {
-      size: 0,
-      query: {
-        bool: {
-          // prefer sampled transactions
-          should: [{ term: { [TRANSACTION_SAMPLED]: true } }],
-        },
-      },
       aggs: {
         transaction_groups: {
-          composite: {
-            size: bucketSize + 1, // 1 extra bucket is added to check whether the total number of buckets exceed the specified bucket size.
```

Comment:
We should retain the comment to explain the `+ 1`

Reply:
there's a comment lower in the file as well, do you think we need both? (no strong opinions from my side)

Reply:
re-added the comment